lib/auth/usertoken_test.go

```diff
@@ -343,6 +350,165 @@ func TestBackwardsCompForUserTokenWithLegacyPrefix(t *testing.T) {
 	require.True(t, trace.IsNotFound(err))
 }
 
+func TestCreatePrivilegeToken(t *testing.T) {
+	t.Parallel()
+	srv := newTestTLSServer(t)
+	fakeClock := srv.Clock().(clockwork.FakeClock)
+	mockEmitter := &events.MockEmitter{}
+	srv.Auth().emitter = mockEmitter
+	ctx := context.Background()
+
+	// Create a user and client with identity.
+	username := "joe@example.com"
+	_, _, err := CreateUserAndRoleWithoutRoles(srv.Auth(), username, []string{username})
+	require.NoError(t, err)
+	clt, err := srv.NewClient(TestUser(username))
+	require.NoError(t, err)
+
+	// Test a failure when second factor isn't enabled.
+	_, err = clt.CreatePrivilegeToken(ctx, &proto.CreatePrivilegeTokenRequest{})
+	require.True(t, trace.IsAccessDenied(err))
+
+	// Enable second factor.
+	ap, err := types.NewAuthPreference(types.AuthPreferenceSpecV2{
+		Type:         constants.Local,
+		SecondFactor: constants.SecondFactorOTP,
+	})
+	require.NoError(t, err)
+	err = srv.Auth().SetAuthPreference(ctx, ap)
+	require.NoError(t, err)
+
+	tests := []struct {
+		name      string
+		tokenType string
+		getReq    func() *proto.CreatePrivilegeTokenRequest
+	}{
+		{
+			name:      "privilege exception token",
+			tokenType: UserTokenTypePrivilegeException,
+			getReq: func() *proto.CreatePrivilegeTokenRequest {
+				return &proto.CreatePrivilegeTokenRequest{
+					ExistingMFAResponse: &proto.MFAAuthenticateResponse{},
```

Comment:
Would a response with a nil `ExistingMFAResponse` fail? Setting an empty ExistingMFAResponse doesn't seem that useful, could we allow nil to work just as empty does?